ppa/archive/admin.py

```diff
@@ -27,6 +31,22 @@ def save_related(self, request, form, formsets, change):
         digwork = DigitizedWork.objects.get(id=form.instance.pk)
         digwork.index(params={"commitWithin": 10000})
 
+    def bulk_add_collection(self, request, queryset):
+        '''
+        Bulk add a queryset of :class:ppa.archive.DigitizedWork to
+        a :class:ppa.archive.Collection.
+        '''
+        # Uses POST from admin rather than a database query to get the pks
```

Comment:
I see that in the examples, although I don't see it as a suggested best practice anywhere (so a link could be useful, if there's a good one to provide).

However, I did some testing and it looks like using the selected value rather than the queryset means you don't get extra items beyond the current page if the user clicks the 'select all' option.  I think supporting that would be valuable, because if admins can search on something that will find a large number of items that belong in a collection, they can add them all at once.

Reply:
I think in our use case, as you say, the queryset is the preferable way to go. It seemed better to avoid the database ping, but given the need to select results beyond 100 (or whatever upper display limit we configure for the admin) and the fact that the admin action misses them currently, that falls behind the need to do something useful.